spec/models/planning_application_spec.rb

```diff
@@ -39,31 +39,235 @@
     end
   end
 
-  describe "statuses" do
-    it "has a list of statuses" do
-      expect(described_class.statuses).to eq(
-        "in_assessment" => 0, "awaiting_determination" => 1, "awaiting_correction" => 2, "determined" => 3
-      )
+  describe "state transitions" do
+    let!(:proposed_drawing_1) do
+      create :drawing, :with_plan, :proposed_tags,
+             planning_application: subject,
+             numbers: "number"
     end
-  end
 
-  describe "update_and_timestamp_status" do
-    described_class.statuses.keys.each do |status|
-      context "for the #{status} status" do
-        before do
-          # Set timestamp to differentiate from now
-          subject.update("#{status}_at": 1.hour.ago)
+    context "start the application" do
+      subject { create :planning_application, :not_started }
+
+      before do
+        # Set timestamp to differentiate from now
+        subject.update("in_assessment_at": 1.hour.ago)
+
+        subject.start
+      end
+
+      it "sets the status to in_assessment" do
+        expect(subject.status).to eq "in_assessment"
+      end
+
+      it "sets the timestamp for in_assessment_at to now" do
+        expect(subject.send("in_assessment_at")).to be_within(1.second).of(Time.current)
+      end
+    end
+
+    context "return the application from invalidated" do
+      subject { create :planning_application, :invalidated }
+
+      before do
+        # Set timestamp to differentiate from now
+        subject.update("returned_at": 1.hour.ago)
+
+        subject.return
+      end
+
+      it "sets the status to returned" do
+        expect(subject.status).to eq "returned"
+      end
+
+      it "sets the timestamp for returned_at to now" do
+        expect(subject.send("returned_at")).to be_within(1.second).of(Time.current)
```

Comment:
I would much prefer to use [Timecop](https://github.com/travisjeffery/timecop) or [ActiveSupport::Testing::TimeHelpers](https://api.rubyonrails.org/v5.2.4/classes/ActiveSupport/Testing/TimeHelpers.html) to fix the time during these tests, rather than assuming they will run within a second.

Reply:
I've updated to use TimeHelpers to avoid third-party dependencies.